terminus/geometry/arc.py

```diff
@@ -88,8 +109,34 @@ def find_intersection(self, other):
     def _find_arc_intersection(self, other):
         circle1 = Circle(self.center_point(), self.radius())
         circle2 = Circle(other.center_point(), other.radius())
-        candidates = circle1.intersection(circle2)
-        return filter(lambda point: self.includes_point(point) and other.includes_point(point), candidates)
+        if (circle1.center - circle2.center).norm() > 1e-5 or abs(circle1.radius - circle2.radius) > 1e-5:
+            candidates = circle1.intersection(circle2)
+            return filter(lambda point: self.includes_point(point) and other.includes_point(point), candidates)
+        else:
+            arc1 = self.counter_clockwise()
+            arc2 = other.counter_clockwise()
+            if arc1 == arc2:
+                return [arc1]
+            elif arc1.end_point().distance_to(arc2.start_point()) < 1e-5 and arc2.end_point().distance_to(arc1.start_point()) < 1e-5:
+                return [arc1.end_point(), arc1.start_point()]
+            elif arc1.end_point().distance_to(arc2.start_point()) < 1e-5:
+                return [arc2.start_point()]
+            elif arc2.end_point().distance_to(arc1.start_point()) < 1e-5:
+                return [arc1.start_point()]
+            elif arc1.includes_point(arc2.start_point()) and arc1.includes_point(arc2.end_point()):
+                if arc2.includes_point(arc1.start_point()) and arc2.includes_point(arc1.end_point()):
+                    return [Arc.from_points_in_circle(arc2.start_point(), arc1.end_point(), circle1),
+                            Arc.from_points_in_circle(arc1.start_point(), arc2.end_point(), circle1)]
+                else:
+                    return [arc2]
+            elif arc1.includes_point(arc2.start_point()):
+                return [Arc.from_points_in_circle(arc2.start_point(), arc1.end_point(), circle1)]
+            elif arc1.includes_point(arc2.end_point()):
+                return [Arc.from_points_in_circle(arc1.start_point(), arc2.end_point(), circle1)]
+            elif arc2.includes_point(arc1.start_point()):
+                return [arc1]
+            else:
+                return []
```

Comment:
Maybe you can extract a couple of other methods from here. For example `arc.contains_arc(other_arc)` which returns `True` if the `other_arc` is fully contained inside `arc`. This is an example of what we talked today, about how to avoid having this big nested `if`s by separating cases early on. So you could simplify things by doing:

```
if self.contains_arc(other_arc):
    return [other_arc]
if other_arc.contains(self):
    return [self]
```

Note that with the above you also solved the issue where arcs are equal.